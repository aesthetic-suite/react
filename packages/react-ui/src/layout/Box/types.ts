import { ShadowSize } from '@aesthetic/react';
import { SpacingProps } from '../../types/spacing';

export type BoxAlignContent =
	| 'baseline'
	| 'center'
	| 'end'
	| 'flex-end'
	| 'flex-start'
	| 'normal'
	| 'safe'
	| 'space-around'
	| 'space-between'
	| 'space-evenly'
	| 'start'
	| 'stretch'
	| 'unsafe';

export type BoxAlignItems =
	| 'baseline'
	| 'center'
	| 'end'
	| 'flex-end'
	| 'flex-start'
	| 'normal'
	| 'safe'
	| 'self-end'
	| 'self-start'
	| 'start'
	| 'stretch'
	| 'unsafe';

export type BoxAlignSelf =
	| 'auto'
	| 'baseline'
	| 'center'
	| 'flex-end'
	| 'flex-start'
	| 'normal'
	| 'safe'
	| 'self-end'
	| 'self-start'
	| 'stretch'
	| 'unsafe';

export type BoxDirection = 'column-reverse' | 'column' | 'row-reverse' | 'row';

export type BoxJustifyContent =
	| 'baseline'
	| 'center'
	| 'end'
	| 'flex-end'
	| 'flex-start'
	| 'left'
	| 'normal'
	| 'right'
	| 'safe'
	| 'space-around'
	| 'space-between'
	| 'space-evenly'
	| 'start'
	| 'unsafe';

export interface BoxProps extends SpacingProps {
	/**
	 * Content to render within the box. Each child will be treated as a flex item.
	 */
	children?: React.ReactNode;
	/**
	 * Define how the browser distributes space between and around content along the cross-axis.
	 * @default normal
	 */
	alignContent?: BoxAlignContent;
	/**
	 * Define how direct children are individually aligned on the cross-axis.
	 * @default normal
	 */
	alignItems?: BoxAlignItems;
	/**
	 * Override this element's cross-axis alignment in its parent flex container.
	 * @default auto
	 */
	alignSelf?: BoxAlignSelf;
	/**
	 * Control the direction of content on the main-axis.
	 * @default row
	 */
	direction?: BoxDirection;
	/**
	 * Set this element's flex grow factor.
	 */
	grow?: number;
	/**
	 * Render box as inline-flex instead of flex.
	 */
	inline?: boolean;
	/**
	 * Define how the browser distributes space between and around content along the main-axis.
	 * @default flex-start
	 */
	justifyContent?: BoxJustifyContent;
	/**
	 * Set the order in which this element appears in its parent flex container.
	 */
	order?: number;
	/**
	 * Apply a shadow with the defined size.
	 */
	shadow?: ShadowSize;
	/**
	 * Set this element's flex shrink factor.
	 */
	shrink?: number;
	/**
	 * Control whether content wraps to another line.
	 * @default false
	 */
	wrap?: boolean | 'reverse';
}
