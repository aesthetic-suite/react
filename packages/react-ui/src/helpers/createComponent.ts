/* eslint-disable @typescript-eslint/no-explicit-any */

import { forwardRef } from 'react';
import { CommonHtmlProps, HtmlElementType, OmitUnwantedHtmlProps } from '../types';

export function createComponent<Props extends object, Element extends HTMLElement = HTMLElement>(
	component: React.ForwardRefRenderFunction<Element, Props>,
) {
	return forwardRef(component);
}

export type DynamicProps<
	As extends React.ElementType,
	Props extends object
> = OmitUnwantedHtmlProps<React.ComponentPropsWithRef<As>> &
	Props & {
		/**
		 * Native HTML element or custom component to render as.
		 */
		as?: As;
	};

// This is based on the `React.ExoticComponent` but supports the dynamic `as` prop
// through a generic on the render function. We want the props on the consumer to change
// based on the `as` prop, but we *do not* want to type the props/ref in the implementation,
// as it makes the types too complex and we spread them into the DOM regardless.
export interface DynamicComponent<Props extends object, As extends React.ElementType> {
	<T extends As>(props: DynamicProps<T, Props>): React.ReactElement | null;
	readonly $$typeof: symbol;
	displayName?: string;
}

// We only want to return generic attributes when an HTML element,
// otherwise return the props from the custom component.
export type InferAsProps<T> = T extends HtmlElementType
	? CommonHtmlProps
	: T extends React.JSXElementConstructor<infer P>
	? P
	: {};

export function createDynamicComponent<
	Props extends object,
	As extends React.ElementType = HtmlElementType
>(component: React.ForwardRefRenderFunction<any, InferAsProps<As> & Props & { as?: As }>) {
	return (createComponent(component) as unknown) as DynamicComponent<Props, As>;
}
