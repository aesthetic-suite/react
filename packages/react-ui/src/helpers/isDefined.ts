export function isDefined(...values: unknown[]): boolean {
	return values.some((value) => typeof value !== 'undefined' && value !== null && value !== false);
}
