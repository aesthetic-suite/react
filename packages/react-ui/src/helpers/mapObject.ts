import { objectLoop } from '@aesthetic/utils';

export function mapObject<P, N>(
  base: Record<string, P>,
  onValue: (value: P) => N | undefined,
  onKey?: (key: string) => string,
): Record<string, N> {
  const object: Record<string, unknown> = {};

  objectLoop(base, (value, key) => {
    const nextValue = onValue(value);

    if (nextValue !== undefined) {
      object[onKey ? onKey(key) : key] = nextValue;
    }
  });

  return object as Record<string, N>;
}
