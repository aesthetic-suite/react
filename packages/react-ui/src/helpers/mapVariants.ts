import { ElementStyles } from '@aesthetic/react';
import { arrayLoop } from '@aesthetic/utils';

export function mapVariants<T>(
	type: string,
	enums: T[],
	factory: (value: T, typeName: string) => ElementStyles,
) {
	const object: Partial<Record<string, ElementStyles>> = {};

	arrayLoop(enums, (value) => {
		object[`${type}:${value}`] = factory(value, type);
	});

	return object as Record<string, ElementStyles>;
}
