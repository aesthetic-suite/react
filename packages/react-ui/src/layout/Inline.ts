import { createItemsContainer, ItemsContainerProps } from './internal/createItemsContainer';

export type InlineProps = ItemsContainerProps;

export const Inline = createItemsContainer<InlineProps>('Inline', 'row');
