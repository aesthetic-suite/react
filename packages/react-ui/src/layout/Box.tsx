import React, { useMemo } from 'react';
import { ShadowSize, useStyles } from '@aesthetic/react';
import { createDynamicComponent } from '../helpers/createComponent';
import { useShadow } from '../hooks/styles/useShadow';
import { useSpacing } from '../hooks/styles/useSpacing';
import { SpacingProps } from '../types/spacing';
import { styleSheet } from './styles';
import {
	BoxAlignContent,
	BoxAlignItems,
	BoxAlignSelf,
	BoxDirection,
	BoxElement,
	BoxJustifyContent,
} from './types';

export * from './types';

export interface BoxProps extends SpacingProps {
	/**
	 * Content to render within the box. Each child will be treated as a flex item.
	 */
	children?: React.ReactNode;
	/**
	 * Define how the browser distributes space between and around content along the cross-axis.
	 * @default normal
	 */
	alignContent?: BoxAlignContent;
	/**
	 * Define how direct children are individually aligned on the cross-axis.
	 * @default normal
	 */
	alignItems?: BoxAlignItems;
	/**
	 * Override this element's cross-axis alignment in its parent flex container.
	 * @default auto
	 */
	alignSelf?: BoxAlignSelf;
	/**
	 * Control the direction of content on the main-axis.
	 * @default row
	 */
	direction?: BoxDirection;
	/**
	 * Set this element's flex grow factor.
	 */
	grow?: number;
	/**
	 * Render box as inline-flex instead of flex.
	 */
	inline?: boolean;
	/**
	 * Define how the browser distributes space between and around content along the main-axis.
	 * @default flex-start
	 */
	justifyContent?: BoxJustifyContent;
	/**
	 * Set the order in which this element appears in its parent flex container.
	 */
	order?: number;
	/**
	 * Apply a shadow with the defined size.
	 */
	shadow?: ShadowSize;
	/**
	 * Set this element's flex shrink factor.
	 */
	shrink?: number;
	/**
	 * Control whether content wraps to another line.
	 * @default false
	 */
	wrap?: boolean | 'reverse';
}

/**
 * @link https://developer.mozilla.org/en-US/docs/Web/Guide/HTML/Content_categories#sectioning_content
 */
export const Box = createDynamicComponent<BoxProps, BoxElement>(function Box(
	{
		alignContent = 'normal',
		alignItems = 'normal',
		alignSelf = 'auto',
		as: Tag = 'div',
		children,
		direction = 'row',
		grow,
		inline = false,
		justifyContent = 'flex-start',
		order,
		shadow,
		shrink,
		wrap = false,
		// Spacing
		spacing,
		spacingBottom,
		spacingEnd,
		spacingHorizontal,
		spacingStart,
		spacingTop,
		spacingVertical,
		// Inherited
		className: inheritedClassName,
		style: inheritedStyle,
		...props
	},
	ref,
) {
	const cx = useStyles(styleSheet);
	const { className: spacingClassName, style: spacingStyle } = useSpacing({
		spacing,
		spacingBottom,
		spacingEnd,
		spacingHorizontal,
		spacingStart,
		spacingTop,
		spacingVertical,
	});
	const shadowClassName = useShadow(shadow);
	const className = useMemo(
		() =>
			cx(
				{ alignContent, alignItems, alignSelf, direction, justifyContent, wrap: String(wrap) },
				'box',
				'boxVars',
				inline && 'boxInline',
				[spacingClassName, shadowClassName, inheritedClassName],
			),
		[
			cx,
			alignContent,
			alignItems,
			alignSelf,
			direction,
			inline,
			justifyContent,
			wrap,
			spacingClassName,
			shadowClassName,
			inheritedClassName,
		],
	);
	const style: React.CSSProperties = useMemo(
		() => ({
			...spacingStyle,
			'--box-grow': grow,
			'--box-order': order,
			'--box-shrink': shrink,
			...inheritedStyle,
		}),
		[grow, order, shrink, spacingStyle, inheritedStyle],
	);

	return (
		<Tag ref={ref} {...props} className={className} style={style}>
			{children}
		</Tag>
	);
});
