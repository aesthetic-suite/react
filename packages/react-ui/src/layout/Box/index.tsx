import React, { useMemo } from 'react';
import { useStyles } from '@aesthetic/react';
import { createDynamicComponent } from '../../helpers/createComponent';
import { useShadow } from '../../hooks/styles/useShadow';
import { useSpacing } from '../../hooks/styles/useSpacing';
import { styleSheet } from './styles';
import { BoxProps } from './types';

export * from './types';

/**
 * @link https://developer.mozilla.org/en-US/docs/Web/Guide/HTML/Content_categories#sectioning_content
 */
export const Box = createDynamicComponent<BoxProps>(function Box(
	{
		alignContent = 'normal',
		alignItems = 'normal',
		alignSelf = 'auto',
		as: Tag = 'div',
		children,
		direction = 'row',
		grow,
		inline = false,
		justifyContent = 'flex-start',
		order,
		shadow,
		shrink,
		wrap = false,
		// Spacing
		spacing,
		spacingBottom,
		spacingEnd,
		spacingHorizontal,
		spacingStart,
		spacingTop,
		spacingVertical,
		// Inherited
		className: inheritedClassName,
		style: inheritedStyle,
		...props
	},
	ref,
) {
	const cx = useStyles(styleSheet);
	const { className: spacingClassName, style: spacingStyle } = useSpacing({
		spacing,
		spacingBottom,
		spacingEnd,
		spacingHorizontal,
		spacingStart,
		spacingTop,
		spacingVertical,
	});
	const shadowClassName = useShadow(shadow);
	const className = useMemo(
		() =>
			cx(
				{ alignContent, alignItems, alignSelf, direction, justifyContent, wrap: String(wrap) },
				'box',
				'vars',
				inline && 'inline',
				[spacingClassName, shadowClassName, inheritedClassName],
			),
		[
			cx,
			alignContent,
			alignItems,
			alignSelf,
			direction,
			inline,
			justifyContent,
			wrap,
			spacingClassName,
			shadowClassName,
			inheritedClassName,
		],
	);
	const style: React.CSSProperties = useMemo(
		() => ({
			...spacingStyle,
			'--box-grow': grow,
			'--box-order': order,
			'--box-shrink': shrink,
			...inheritedStyle,
		}),
		[grow, order, shrink, spacingStyle, inheritedStyle],
	);

	return (
		<Tag ref={ref} {...props} className={className} style={style}>
			{children}
		</Tag>
	);
});
