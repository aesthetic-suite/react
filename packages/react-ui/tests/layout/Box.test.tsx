import React from 'react';
import { hyphenate } from '@aesthetic/utils';
import { render, screen } from '@testing-library/react';
import { Box, BoxProps } from '../../src/layout/Box';
import { getRenderedStyles, withStyles, Wrapper } from '../helpers';

function getElement() {
	return screen.getByTestId('box')!;
}

describe('Box', () => {
	it('renders the box', () => {
		render(<Box testID="box">Child</Box>, { wrapper: Wrapper });

		expect(getElement().className).toBe(
			'box variant:align-content:normal variant:align-items:normal variant:align-self:auto variant:direction:row variant:justify-content:flex-start variant:wrap:false element element',
		);
	});

	it(
		'renders the box styles',
		withStyles(() => {
			render(<Box testID="box">Child</Box>, { wrapper: Wrapper });

			expect(getRenderedStyles('standard')).toMatchSnapshot();
		}),
	);

	it('can change the element tag', () => {
		render(
			<Box as="article" testID="box">
				Child
			</Box>,
			{ wrapper: Wrapper },
		);

		expect(getElement().tagName).toBe('ARTICLE');
	});

	it('can pass a test ID', () => {
		render(<Box testID="box">Child</Box>, { wrapper: Wrapper });

		expect(getElement().getAttribute('data-testid')).toBe('box');
	});

	it('can pass a custom class name', () => {
		render(
			<Box className="foobar" testID="box">
				Title
			</Box>,
			{ wrapper: Wrapper },
		);

		expect(getElement().className).toEqual(expect.stringContaining('foobar'));
	});

	it('can pass custom styles', () => {
		render(
			<Box style={{ color: 'red' }} testID="box">
				Title
			</Box>,
			{ wrapper: Wrapper },
		);

		expect(getElement().style.color).toBe('red');
	});

	it('can pass native attributes', () => {
		render(
			<Box aria-label="Label" id="foo" testID="box">
				Title
			</Box>,
			{ wrapper: Wrapper },
		);

		const el = getElement();

		expect(el.id).toBe('foo');
		expect(el.getAttribute('aria-label')).toBe('Label');
	});

	describe('props', () => {
		const props: BoxProps = {
			alignContent: 'space-around',
			alignItems: 'safe',
			alignSelf: 'center',
			direction: 'column',
			justifyContent: 'space-between',
			wrap: 'reverse',
		};

		Object.entries(props).forEach(([prop, value]) => {
			it(`sets "${prop}" class name`, () => {
				render(
					<Box testID="box" {...{ [prop]: value }}>
						Child
					</Box>,
					{ wrapper: Wrapper },
				);

				expect(getElement().className).toEqual(
					expect.stringContaining(`variant:${hyphenate(prop)}:${value}`),
				);
			});
		});

		it('sets "inline" class name', () => {
			render(
				<Box inline testID="box">
					Child
				</Box>,
				{ wrapper: Wrapper },
			);

			expect(getElement().className).toEqual(expect.stringContaining('boxInline'));
		});

		it('sets "grow" class name and style variable', () => {
			render(
				<Box grow={3} testID="box">
					Child
				</Box>,
				{ wrapper: Wrapper },
			);

			expect(getElement().className).toEqual(expect.stringContaining('boxVars'));
			expect(getElement().style.getPropertyValue('--box-grow')).toBe('3');
		});

		it('sets "shrink" class name and style variable', () => {
			render(
				<Box shrink={2} testID="box">
					Child
				</Box>,
				{ wrapper: Wrapper },
			);

			expect(getElement().className).toEqual(expect.stringContaining('boxVars'));
			expect(getElement().style.getPropertyValue('--box-shrink')).toBe('2');
		});

		it('sets "order" class name and style variable', () => {
			render(
				<Box order={0} testID="box">
					Child
				</Box>,
				{ wrapper: Wrapper },
			);

			expect(getElement().className).toEqual(expect.stringContaining('boxVars'));
			expect(getElement().style.getPropertyValue('--box-order')).toBe('0');
		});

		it('sets spacing style variables', () => {
			render(
				<Box spacing={1} spacingHorizontal={2} testID="box">
					Child
				</Box>,
				{ wrapper: Wrapper },
			);

			expect(getElement().style.getPropertyValue('--spacing-all')).toBe('1.25rem');
			expect(getElement().style.getPropertyValue('--spacing-horizontal')).toBe('2.50rem');
		});
	});
});
