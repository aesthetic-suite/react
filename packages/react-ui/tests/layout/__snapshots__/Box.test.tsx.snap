// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box renders the box styles 1`] = `".c4n8yy0 {display: flex;}.c13kbekr {margin: 0;}.cqmlbj8 {padding: 0;}.c1ht7y6v {list-style: none;}.c1fh3frr {align-content: baseline;}.ckqeief {align-content: center;}.cwlwkhv {align-content: end;}.c17s6ji1 {align-content: flex-end;}.c2pa7sm {align-content: flex-start;}.cuo9hcv {align-content: normal;}.cr3fhy5 {align-content: safe;}.c10k3ayu {align-content: space-around;}.c1iss4bz {align-content: space-between;}.c1eglky0 {align-content: space-evenly;}.c1rhwx4c {align-content: start;}.c7xm1c3 {align-content: stretch;}.c1kvv06e {align-content: unsafe;}.c1mcbrfc {align-items: baseline;}.c15h9hk8 {align-items: center;}.c1mnp32k {align-items: end;}.cxqk6u {align-items: flex-end;}.cfhcfi1 {align-items: flex-start;}.c4c6974 {align-items: normal;}.cattwnm {align-items: safe;}.c1hbgmgt {align-items: self-end;}.cy5rxiq {align-items: self-start;}.ct16uqr {align-items: start;}.cad9k5o {align-items: stretch;}.czhj38p {align-items: unsafe;}.c11sa73q {align-self: auto;}.c50su02 {align-self: baseline;}.c548toy {align-self: center;}.c8d6o3w {align-self: flex-end;}.c1rf4m77 {align-self: flex-start;}.ce5nrze {align-self: normal;}.c112fors {align-self: safe;}.ccy8iv {align-self: self-end;}.c14c7jgo {align-self: self-start;}.ciuyth2 {align-self: stretch;}.cvrvfsj {align-self: unsafe;}.cdx6gi4 {flex-direction: column;}.c3ceald {flex-direction: column-reverse;}.co6ao34 {flex-direction: row;}.cdgl4u5 {flex-direction: row-reverse;}.c1c51rl0 {justify-content: center;}.cnceuww {justify-content: end;}.c1rpm0l6 {justify-content: flex-end;}.cbgwkvp {justify-content: flex-start;}.covi1dg {justify-content: left;}.c1k5r2ak {justify-content: normal;}.c14w7du7 {justify-content: right;}.c1kacjj2 {justify-content: safe;}.c1njp0xx {justify-content: space-around;}.cfi6kh8 {justify-content: space-between;}.c8o2b63 {justify-content: space-evenly;}.c1fol5rj {justify-content: start;}.c1lk4411 {justify-content: unsafe;}.c193j6ji {flex-wrap: nowrap;}.c164ecv3 {flex-wrap: wrap;}.c9n48wi {flex-wrap: wrap-reverse;}.ceq1ljg {display: inline-flex;}.c14t3bnn {flex-grow: var(--box-grow);}.c1u1jhtv {flex-shrink: var(--box-shrink);}.c1636tmh {order: var(--box-order);}.c15t3j1l> *:not(:last-child) {margin-right: var(--gap);}.cx3eaue> *:not(:last-child) {margin-bottom: var(--gap);}.c1g5rurh {padding: var(--spacing-all);}.cibiy7j {padding: var(--spacing-xxs);}.cj52zuf {padding: var(--spacing-xs);}.cfxizb6 {padding: var(--spacing-sm);}.c1xb6ewu {padding: var(--spacing-df);}.c1ugvpp1 {padding: var(--spacing-md);}.c15ovmnr {padding: var(--spacing-lg);}.cyup0mg {padding: var(--spacing-xl);}.c1b71xjk {padding: var(--spacing-xxl);}.cw2ew7o {padding-left: var(--spacing-horizontal);}.c1yx0hov {padding-right: var(--spacing-horizontal);}.c54gyhl {padding-left: var(--spacing-xxs);}.c1rv52ci {padding-right: var(--spacing-xxs);}.c11b8tkx {padding-left: var(--spacing-xs);}.c1pyn0u2 {padding-right: var(--spacing-xs);}.c2n2las {padding-left: var(--spacing-sm);}.c1oyd0wf {padding-right: var(--spacing-sm);}.cfnz2qw {padding-left: var(--spacing-df);}.c1mj2ayb {padding-right: var(--spacing-df);}.cjyu3z7 {padding-left: var(--spacing-md);}.cv7x7q0 {padding-right: var(--spacing-md);}.c1nv22ip {padding-left: var(--spacing-lg);}.cos7m2i {padding-right: var(--spacing-lg);}.c11mjqpq {padding-left: var(--spacing-xl);}.c12aiuhx {padding-right: var(--spacing-xl);}.c14crmme {padding-left: var(--spacing-xxl);}.cjgzuzh {padding-right: var(--spacing-xxl);}.cplmy08 {padding-top: var(--spacing-vertical);}.c13z83ws {padding-bottom: var(--spacing-vertical);}.c1bowjs9 {padding-top: var(--spacing-xxs);}.cefgwml {padding-bottom: var(--spacing-xxs);}.c1yh1hbl {padding-top: var(--spacing-xs);}.c1beclth {padding-bottom: var(--spacing-xs);}.cwdul6s {padding-top: var(--spacing-sm);}.c16ip6ow {padding-bottom: var(--spacing-sm);}.c5vcj94 {padding-top: var(--spacing-df);}.cy689wc {padding-bottom: var(--spacing-df);}.c17rw3f7 {padding-top: var(--spacing-md);}.cglihxz {padding-bottom: var(--spacing-md);}.cavyxdd {padding-top: var(--spacing-lg);}.cs15051 {padding-bottom: var(--spacing-lg);}.c1atx52m {padding-top: var(--spacing-xl);}.c11w6nd6 {padding-bottom: var(--spacing-xl);}.cnhixp2 {padding-top: var(--spacing-xxl);}.cqtbhki {padding-bottom: var(--spacing-xxl);}.c1f31uz5 {padding-top: var(--spacing-top);}.c1bowjs9 {padding-top: var(--spacing-xxs);}.c1yh1hbl {padding-top: var(--spacing-xs);}.cwdul6s {padding-top: var(--spacing-sm);}.c5vcj94 {padding-top: var(--spacing-df);}.c17rw3f7 {padding-top: var(--spacing-md);}.cavyxdd {padding-top: var(--spacing-lg);}.c1atx52m {padding-top: var(--spacing-xl);}.cnhixp2 {padding-top: var(--spacing-xxl);}.c1o3yudt {padding-bottom: var(--spacing-bottom);}.cefgwml {padding-bottom: var(--spacing-xxs);}.c1beclth {padding-bottom: var(--spacing-xs);}.c16ip6ow {padding-bottom: var(--spacing-sm);}.cy689wc {padding-bottom: var(--spacing-df);}.cglihxz {padding-bottom: var(--spacing-md);}.cs15051 {padding-bottom: var(--spacing-lg);}.c11w6nd6 {padding-bottom: var(--spacing-xl);}.cqtbhki {padding-bottom: var(--spacing-xxl);}.c1yeaeq2 {padding-left: var(--spacing-start);}.c54gyhl {padding-left: var(--spacing-xxs);}.c11b8tkx {padding-left: var(--spacing-xs);}.c2n2las {padding-left: var(--spacing-sm);}.cfnz2qw {padding-left: var(--spacing-df);}.cjyu3z7 {padding-left: var(--spacing-md);}.c1nv22ip {padding-left: var(--spacing-lg);}.c11mjqpq {padding-left: var(--spacing-xl);}.c14crmme {padding-left: var(--spacing-xxl);}.cdpbbi6 {padding-right: var(--spacing-end);}.c1rv52ci {padding-right: var(--spacing-xxs);}.c1pyn0u2 {padding-right: var(--spacing-xs);}.c1oyd0wf {padding-right: var(--spacing-sm);}.c1mj2ayb {padding-right: var(--spacing-df);}.cv7x7q0 {padding-right: var(--spacing-md);}.cos7m2i {padding-right: var(--spacing-lg);}.c12aiuhx {padding-right: var(--spacing-xl);}.cjgzuzh {padding-right: var(--spacing-xxl);}.c15c9dnj {box-shadow: var(--shadow-xs-x) var(--shadow-xs-y) var(--shadow-xs-blur) var(--shadow-xs-spread) rgba(0, 0, 0, .8);}.co1m4cf {box-shadow: var(--shadow-sm-x) var(--shadow-sm-y) var(--shadow-sm-blur) var(--shadow-sm-spread) rgba(0, 0, 0, .8);}.c7d7xjz {box-shadow: var(--shadow-md-x) var(--shadow-md-y) var(--shadow-md-blur) var(--shadow-md-spread) rgba(0, 0, 0, .8);}.cm94rxb {box-shadow: var(--shadow-lg-x) var(--shadow-lg-y) var(--shadow-lg-blur) var(--shadow-lg-spread) rgba(0, 0, 0, .8);}.c18d7en {box-shadow: var(--shadow-xl-x) var(--shadow-xl-y) var(--shadow-xl-blur) var(--shadow-xl-spread) rgba(0, 0, 0, .8);}"`;
